modules/dxp/apps/analytics/analytics-reports-web/src/test/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProviderTest.java

```diff
@@ -422,6 +423,168 @@ public void testGetTrafficChannelsWithCountrySearchKeywords()
 			String.valueOf(trafficChannels.get("paid")));
 	}
 
+	@Test
+	public void testGetTrafficSources() throws Exception {
+		AnalyticsReportsDataProvider analyticsReportsDataProvider =
+			new AnalyticsReportsDataProvider(
+				_getHttp(
+					Collections.singletonMap(
+						"/traffic-sources",
+						JSONUtil.putAll(
+							JSONUtil.put(
+								"name", "organic"
+							).put(
+								"trafficAmount", 1192L
+							).put(
+								"trafficShare", 83.9D
+							),
+							JSONUtil.put(
+								"name", "paid"
+							).put(
+								"trafficAmount", 2010L
+							).put(
+								"trafficShare", 44.1D
+							)
+						).toString())));
+
+		Map<String, TrafficSource> trafficSources =
+			analyticsReportsDataProvider.getTrafficSources(
+				RandomTestUtil.randomLong(), RandomTestUtil.randomString());
+
+		Assert.assertEquals(
+			trafficSources.toString(), 2, trafficSources.size());
+		Assert.assertEquals(
+			String.valueOf(
+				new TrafficSource(
+					Collections.emptyList(), "organic", 1192L, 83.9D)),
+			String.valueOf(trafficSources.get("organic")));
+		Assert.assertEquals(
+			String.valueOf(
+				new TrafficSource(
+					Collections.emptyList(), "paid", 2010L, 44.1D)),
+			String.valueOf(trafficSources.get("paid")));
+	}
+
+	@Test
+	public void testGetTrafficSourcesWithAsahFaroBackendError()
```

Comment:
Hey @cgoncas, I decided to leave the exception handling as it was to not change the current behavior, let me know what you think. Thx!